autoload/asyncomplete/sources/tmuxcomplete.vim

```diff
@@ -0,0 +1,38 @@
+function! asyncomplete#sources#tmuxcomplete#register(opts)
+    let l:opts = extend(extend({}, a:opts), {
+                \ 'name':      'tmuxcomplete',
+                \ 'whitelist': ['*'],
+                \ 'completor': function('asyncomplete#sources#tmuxcomplete#completor'),
+                \ })
+    call asyncomplete#register_source(l:opts)
+endfunction
+
+function! asyncomplete#sources#tmuxcomplete#completor(opt, ctx)
+    " taken from asyncomplete-buffer
+    let l:kw = matchstr(a:ctx['typed'], '\w\+$')
+    let a:opt['kwlen'] = len(l:kw)
+    if a:opt['kwlen'] < 1
+        return
+    endif
+
+    let l:cmd = tmuxcomplete#getcommandlist(l:kw, 'words')
+    let a:opt['buffer'] = ''
+
+    let l:jobid = async#job#start(l:cmd, {
+                \ 'on_stdout': function('s:handler', [a:opt, a:ctx]),
```

Comment:
since you are passing your options. you could add `a:ctx` as part of the option.